#
# Submitted:
#
# Marc Kleine-Budde <kleine-budde@gmx.de>, 2005-08-16
#
# Error:
#
# make[1]: Entering directory `/home/frogger/pengutronix/pii/ptxdist-0.7-trunk/build/ntp-4.2.0/ntpdc'
# if gcc -DHAVE_CONFIG_H -I. -I. -I.. -I../include -isystem
# /home/frogger/pengutronix/pii/ptxdist-0.7-trunk/local/arm-softfloat-linux-gnu/include
# -O -Wall -Wcast-qual -Wmissing-prototypes -Wpointer-arith -Wshadow
# -Wstrict-prototypes -MT ntpdc-layout.o -MD -MP -MF
# ".deps/ntpdc-layout.Tpo" \
#   -c -o ntpdc-layout.o `test -f 'ntpdc-layout.c' || echo './'`ntpdc-layout.c; \
# then mv -f ".deps/ntpdc-layout.Tpo" ".deps/ntpdc-layout.Po"; \
# else rm -f ".deps/ntpdc-layout.Tpo"; exit 1; \
# fi
# In file included from ntpdc-layout.c:9:
# ../include/ntp_stdlib.h:138: error: array type has incomplete element type
#
# Description:
#
# problem with gcc-4.x ???
#
# see:
# http://bugs.openembedded.org/show_bug.cgi?id=183
# http://bugs.openembedded.org/attachment.cgi?id=122
#
# State:
#
# this is patch: http://bugs.openembedded.org/attachment.cgi?id=122
#
--- ntp-stable-4.2.0a-20040617/include/ntp_stdlib.h.gcc4	2005-03-08 14:29:42.940559784 +0100
+++ ntp-stable-4.2.0a-20040617/include/ntp_stdlib.h	2005-03-08 14:29:43.015548384 +0100
@@ -133,10 +133,6 @@
 extern u_char *	cache_key;		/* key pointer */
 extern u_int	cache_keylen;		/* key length */
 
-/* clocktypes.c */
-struct clktype;
-extern struct clktype clktypes[];
-
 /* getopt.c */
 extern char *	ntp_optarg;		/* global argument pointer */
 extern int	ntp_optind;		/* global argv index */
--- ntp-stable-4.2.0a-20040617/include/ntp_refclock.h.gcc4	2004-02-25 06:57:56.000000000 +0100
+++ ntp-stable-4.2.0a-20040617/include/ntp_refclock.h	2005-03-08 14:29:43.015548384 +0100
@@ -65,6 +65,7 @@
 	const char *clocktype;	/* long description */
 	const char *abbrev;	/* short description */
 };
+extern struct clktype clktypes[];
 
 /*
  * Configuration flag values
--- ntp-stable-4.2.0a-20040617/include/ntpd.h.gcc4	2004-05-25 13:02:24.000000000 +0200
+++ ntp-stable-4.2.0a-20040617/include/ntpd.h	2005-03-08 14:29:43.016548232 +0100
@@ -225,8 +225,6 @@
 #endif
 
 /* ntp_control.c */
-struct ctl_trap;
-extern struct ctl_trap ctl_trap[];
 extern int	num_ctl_traps;
 extern keyid_t	ctl_auth_keyid;		/* keyid used for authenticating write requests */
 
--- ntp-stable-4.2.0a-20040617/include/ntp_control.h.gcc4	2004-05-25 13:02:23.000000000 +0200
+++ ntp-stable-4.2.0a-20040617/include/ntp_control.h	2005-03-08 14:29:43.016548232 +0100
@@ -268,6 +268,7 @@
 	u_char tr_flags;		/* trap flags */
 	u_char tr_version;		/* version number of trapper */
 };
+extern struct ctl_trap ctl_trap[];
 
 /*
  * Flag bits
